calico/felix/config.py

```diff
@@ -373,6 +385,36 @@ def _validate_cfg(self, final=True):
             raise ConfigException("Invalid port in field",
                                   self.parameters["EtcdAddr"])
 
+        if self.ETCD_SCHEME == "https":
+            # key and certificate must be both specified or both not specified
+            if bool(self.ETCD_KEY_FILE) != bool(self.ETCD_CERT_FILE):
+                raise ConfigException("Invalid (key, certificate) "
+                                      "combination. Key and certificate must "
+                                      "both be specified or both be blank.",
+                                      (self.parameters["EtcdKeyFile"],
+                                       self.parameters["EtcdCertFile"]))
+
+            # Make sure etcd key and certificate are readable
+            if self.ETCD_KEY_FILE and self.ETCD_CERT_FILE:
+                if not (os.access(self.ETCD_KEY_FILE, os.R_OK) and
+                            os.access(self.ETCD_CERT_FILE, os.R_OK)):
+                    raise ConfigException("Cannot read key and/or certificate "
+                                          "file(s). Both must be readable "
+                                          "file paths. ",
+                                          (self.parameters["EtcdKeyFile"],
+                                           self.parameters["EtcdCertFile"]))
+
+                # If Certificate Authority cert provided, check it's readable
+                if self.ETCD_CA_FILE and \
+                        not os.access(self.ETCD_CA_FILE, os.R_OK):
+                    raise ConfigException("Cannot read CA certificate. Value "
+                                          "must be readable file path.",
+                                          self.parameters["EtcdCaFile"])
+        elif self.ETCD_SCHEME != "http":
+            raise ConfigException("Invalid protocol scheme. Value must be one "
+                                  "of: \"\", \"http\", \"https\".",
```

Comment:
If it has reached this point and the environment var was set to `""`, it would interpret this value as not being set and give it the default value of "http".

We should never reach this point with the value being set to `""`, unless I am misinterpreting how `add_parameter` works with defaults for environment variables in config.py.